src/Autodesk.Forge.Core/ForgeHandler.cs

```diff
@@ -31,21 +31,28 @@ namespace Autodesk.Forge.Core
 {
     public class ForgeHandler : DelegatingHandler
     {
-        private const int ForgeGatewayTimeoutSeconds = 10;
-        static SemaphoreSlim semaphore = new SemaphoreSlim(1, 1);
-
+        private static SemaphoreSlim semaphore = new SemaphoreSlim(1, 1);
         private readonly Random rand = new Random();
         private readonly IAsyncPolicy<HttpResponseMessage> resiliencyPolicies;
         protected readonly IOptions<ForgeConfiguration> configuration;
+
         protected ITokenCache TokenCache { get; private set; }
 
         public ForgeHandler(IOptions<ForgeConfiguration> configuration)
         {
             this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
             this.TokenCache = new TokenCache();
-            this.resiliencyPolicies = GetResiliencyPolicies();
+            this.resiliencyPolicies = GetResiliencyPolicies(GetDefaultTimeout());
+        }
+        protected virtual TimeSpan GetDefaultTimeout()
+        {
+             // use timeout greater than the forge gateways (10s), we handle the GatewayTimeout response
+            return TimeSpan.FromSeconds(15); 
+        }
+        protected virtual (int baseDelayInMs, int multiplier ) GetRetryParameters()
+        {
+            return (500, 1000);
         }
```

Comment:
We want to override these in unit tests so that they don't take forever.

Reply:
GitHub can comment on multiple lines!